language/move-prover/boogie-backend-v2/src/options.rs

```diff
@@ -84,7 +84,7 @@ impl Default for BoogieOptions {
             generate_smt: false,
             native_equality: false,
             type_requires: "free requires".to_owned(),
-            stratification_depth: 4,
+            stratification_depth: 6,
```

Comment:
Also wondering why you needed to bump up the stratification depth?

Reply:
Yeah, this is one of the cases where extensional equality hit. It appears we exceeding the previous depth of 4, but this wasn't spotted because luckily, the `==` worked (which we do once we exceed the configured level).

Reply:
As I said, any place where we model a "let" via an assume. Those lets stem from various points in the bytecode transformation. To see them you can find the function `FunctionDataBuilder::emit_let` and then let your IDE search for it's usages. 

As a side note, the current let implementation assumes loops have been removed. It would require a havoc of the lhs if this will not be eventually maintaned.